### ISSUE TITLE
Bug in tables when empty row labels are given

### ISSUE THREAD

**Issue by Melkiades** [insightsengineering/formatters#93]
Found a bug in the following downstream tables: adat03 and pkct01. This is due to having lines with empty labels (" ") which messes up the recognition of the indentation. In the specific this line: `old_indent <- gsub("^([[:space:]]*).*", "\\1", mat$strings[, 1])`

_Originally posted by @Melkiades in https://github.com/insightsengineering/formatters/issues/90#issuecomment-1409235363_

**Comment by gmbecker** [comment]
Good catch. likely the only way around this is to calculate the indents at matrix_form creation time and carry them around in the object one way or another (just the size should be sufficient, we don't actually needmthe strings with empty spaces).

Another check might involve the colwidths if we carry those around, but carrying the final indent sizes around is likely to be cleaner

**Comment by gmbecker** [comment]
In fact we already have this, `mf_rinfo(matform)` is the row info dataframe (ie output of `make_row_df`), which has an `indent` column which is the computed indents (after any indent modifiers), so please refactor the reindentation code to use that in the process of fixing the other bug @Melkiades

**Comment by Melkiades** [comment]
I fixed it by using the `mf_rinfo(matform)$indent` that keeps track perfectly of the indentation, but it was not keeping track of `topleft` indentation. Now it does, with a hybrid solution that assumes no one will add `topleft = " "`. This does not break anything eventually. 

```
 ————————————————————————————————————————————————————————————
  Ethnicity                                 A: Drug X         
  Sex                                  abc            def     
  ————————————————————————————————————————————————————————————
      ASIAN                         79 (65.3%)     79 (65.3%) 
     F                                                        
          asd   
````

**Comment by gmbecker** [comment]
So topleft "indentention" isn't really, its just empty space already in the strings. We coud change that, I suppose, but I don't really see a pressing reason to.

In light of that, I think we just need to check if there is topleft material there, and determine how much leading whitespace it starts with, and restore that exactly.
